savant/utils/re_patterns.py

```diff
@@ -0,0 +1,4 @@
+import re
+
+socket_uri_pattern = re.compile('([a-z]+\\+[a-z]+:)?([a-z]+://.*)')
```

Comment:
Since we expect only a very limited set of possible words, I'd write more concrete regex for the socket options

Something like

(sub|pub|req|rep|etc...)+(sub|pub|req|rep|etc...)

Up to you.